megamek/src/megamek/common/verifier/TestTank.java

```diff
@@ -451,6 +458,71 @@ public boolean correctEntity(StringBuffer buff, int ammoTechLvl) {
+                || mode.equals(EntityMovementMode.HYDROFOIL)
+                || mode.equals(EntityMovementMode.SUBMARINE);
+        if (eq instanceof MiscType) {
+            if (eq.hasFlag(MiscType.F_FLOTATION_HULL)) {
+                // Per errata, WiGE vehicles automatically include flotation hull
+                return mode.equals(EntityMovementMode.HOVER) || mode.equals(EntityMovementMode.VTOL);
+            }
+            if (eq.hasFlag(MiscType.F_FULLY_AMPHIBIOUS)
+                    || eq.hasFlag(MiscType.F_LIMITED_AMPHIBIOUS)
+                    || eq.hasFlag(MiscType.F_BULLDOZER)
+                    || (eq.hasFlag(MiscType.F_CLUB) && eq.hasSubType(MiscType.S_COMBINE))) {
+                return mode.equals(EntityMovementMode.WHEELED) || mode.equals(EntityMovementMode.TRACKED);
+            }
+            if (eq.hasFlag(MiscType.F_DUNE_BUGGY)) {
+                return mode.equals(EntityMovementMode.WHEELED);
+            }
+            if (eq.hasFlag(MiscType.F_CLUB)
+                    && eq.hasSubType(MiscType.S_CHAINSAW | MiscType.S_DUAL_SAW | MiscType.S_MINING_DRILL)) {
+                return mode.equals(EntityMovementMode.WHEELED) || mode.equals(EntityMovementMode.TRACKED)
+                        || mode.equals(EntityMovementMode.HOVER) || mode.equals(EntityMovementMode.WIGE);
+            }
+            if (eq.hasFlag(MiscType.F_MINESWEEPER)) {
+                return mode.equals(EntityMovementMode.WHEELED) || mode.equals(EntityMovementMode.TRACKED)
+                    || isNaval;
+            }
+            if (eq.hasFlag(MiscType.F_LIFEBOAT)) {
+                // Need to filter out atmospheric lifeboat
+                return isNaval && (eq.hasFlag(MiscType.F_TANK_EQUIPMENT) || eq.hasFlag(MiscType.F_SUPPORT_TANK_EQUIPMENT));
+            }
+            if (eq.hasFlag(MiscType.F_HEAVY_BRIDGE_LAYER)
+                    || eq.hasFlag(MiscType.F_MEDIUM_BRIDGE_LAYER)
+                    || eq.hasFlag(MiscType.F_LIGHT_BRIDGE_LAYER)
+                    || (eq.hasFlag(MiscType.F_CLUB)
+                    && eq.hasSubType(MiscType.S_BACKHOE | MiscType.S_ROCK_CUTTER
+                    | MiscType.S_SPOT_WELDER | MiscType.S_WRECKING_BALL))) {
+                return !mode.equals(EntityMovementMode.VTOL);
+            }
+            if (eq.hasFlag(MiscType.F_CLUB) && eq.hasSubType(MiscType.S_PILE_DRIVER)) {
+                return !mode.equals(EntityMovementMode.VTOL)
+                        && !mode.equals(EntityMovementMode.HOVER)
+                        && !mode.equals(EntityMovementMode.WIGE);
+            }
+            if (eq.hasFlag(MiscType.F_AP_POD)) {
+                return !isNaval;
+            }
+        } else if (eq instanceof WeaponType) {
+            if (((WeaponType) eq).getAmmoType() == AmmoType.T_BPOD) {
+                return !isNaval;
+            }
+            if (((WeaponType) eq).getAmmoType() == AmmoType.T_NAIL_RIVET_GUN) {
```

Comment:
What? VTOLs can't mount rivet guns? That's BS, I demand a refund! ::monocle::

Reply:
I think it's a total crock that submarines can't have combines. Hello? Kelp forests?